src/core/eme/eme_manager.ts

```diff
@@ -107,88 +133,89 @@ export default function EMEManager(
    // This is to avoid handling multiple times the same encrypted events.
   const handledInitData = new InitDataStore();
 
-  /* Catch "encrypted" event and create MediaKeys */
-  return observableCombineLatest(
-    onEncrypted$(mediaElement),
-    getMediaKeysInfos(
-      mediaElement,
-      keySystemsConfigs,
-      attachedMediaKeysInfos
-    )
-  ).pipe(
-
-    /* Attach server certificate and create/reuse MediaKeySession */
-    mergeMap(([encryptedEvent, mediaKeysInfos], i) => {
-      log.debug("EME: encrypted event received", encryptedEvent);
-
-      const { keySystemOptions, mediaKeys } = mediaKeysInfos;
-      const { serverCertificate } = keySystemOptions;
-
-      const session$ = getSession(encryptedEvent, handledInitData, mediaKeysInfos)
-        .pipe(map((evt) => ({
-          type: evt.type,
-          value: {
-            initData: evt.value.initData,
-            initDataType: evt.value.initDataType,
-            mediaKeySession: evt.value.mediaKeySession,
-            sessionType: evt.value.sessionType,
-            keySystemOptions: mediaKeysInfos.keySystemOptions,
-            sessionStorage: mediaKeysInfos.sessionStorage,
-          },
-        })));
-
-      if (i === 0) { // first encrypted event for the current content
-        return observableMerge(
-          serverCertificate != null ?
-
-            observableConcat(
-              setServerCertificate(mediaKeys, serverCertificate),
-              session$
-            ) : session$,
-
-            attachMediaKeys(mediaKeysInfos, mediaElement, attachedMediaKeysInfos)
-              .pipe(ignoreElements())
-        );
-      }
-
-      return session$;
+  const initMediaKeys$ = initMediaKeys(
+    mediaElement, keySystemsConfigs, attachedMediaKeysInfos);
```

Comment:
If the encrypted event is received before the mediakeys has been created and attached, we won't be able to decrypt the stream.

Reply:
I still think it would be cleaner to link the event listener first. That way, nothing can go wrong, even if we decide not to wait for the eme-init signal in certain cases in the ``Init`` code.